sdk/include/opentelemetry/sdk/trace/span_data.h

```diff
@@ -225,6 +232,11 @@ class SpanData final : public Recordable
     span_kind_ = span_kind;
   }
 
+  void SetResource(const opentelemetry::sdk::resource::Resource &resource) noexcept override
+  {
+    resource_ = &resource;
```

Comment:
@lalitb - this is problematic spot. It won't work too well with temporaries because it's assigning a pointer to reference. If `Resource` is compiled with `std::variant`, and a temporary is passed to API, e.g. `SetResource("key", "value")` - it'd break..